### PR TITLE
Rentention policy

### PR DESCRIPTION
**Pull request by SwannCroiset**

### CODE REVIEW

File: influxdb/server.sls [hunk 1]
```diff
@@ -77,6 +77,22 @@ influxdb_create_db_{{db.name}}:
...
+
+    {% for rp in db.get('retention_policy', []) %}
+    {% set rp_name = rp.get('name', 'autogen') %}
+    {% set duration = rp.get('duration', 'INF') %}
+    {% set replication = rp.get('replication', '1') %}
+    {% set shard_duration = rp.get('shard_duration', False) %}
+    {% set query_rentention_policy = 'RETENTION POLICY {} ON {} DURATION {} REPLICATION {}'.format(
+        rp_name, db.name, duration, replication)
+    %}
+influxdb_retention_policy_{{db.name}}{{ rp_name }}:
```
Comment (elemoine) [inline influxdb/server.sls]: Do you want to add an underscore between the db name and the rp name in the declaration ID?

File: influxdb/server.sls [hunk 2]
```diff
@@ -77,6 +77,22 @@ influxdb_create_db_{{db.name}}:
       - cmd: influxdb_create_admin
     {%- endif %}
   # TODO: manage database deletion
+
+    {%- if db.get('retention_policy') is mapping %}
+    {% set rp_name = db.retention_policy.get('name', 'autogen') %}
+    {% set duration = db.retention_policy.get('duration', 'INF') %}
+    {% set replication = db.retention_policy.get('replication', '1') %}
+    {% set shard_duration = db.retention_policy.get('shard_duration', False) %}
+    {% set query_rentention_policy = 'RETENTION POLICY {} ON {} DURATION {} REPLICATION {}'.format(
```
Comment (simonpasquier) [inline influxdb/server.sls]: s/query_rentention_policy/query_retention_policy/

File: influxdb/server.sls [hunk 3]
```diff
@@ -77,6 +77,22 @@ influxdb_create_db_{{db.name}}:
...
+        rp_name, db.name, duration, replication)
+    %}
+influxdb_retention_policy_{{db.name}}:
+  cmd.run:
+
+    - name: curl -s -S -POST "{{ url_for_query }}" --data-urlencode "q=CREATE {{ query_rentention_policy }}{%- if shard_duration %} SHARD DURATION {{ shard_duration }}{%- endif %}"|grep "policy already exists" || curl -s -S -POST "{{ url_for_query }}" --data-urlencode "q=ALTER {{ query_rentention_policy }}{%- if shard_duration %} SHARD DURATION {{ shard_duration }}{%- endif %}"
```
Comment (simonpasquier) [inline influxdb/server.sls]: not sure why "SHARD DURATION ..." isn't added to `query_rentention_policy` before.

**Review by simonpasquier — CHANGES_REQUESTED** [review]

File: influxdb/server.sls [hunk 4]
```diff
@@ -77,6 +77,22 @@ influxdb_create_db_{{db.name}}:
...
     {%- endif %}
   # TODO: manage database deletion
+
+    {% for rp in db.get('retention_policy', []) %}
+    {% set rp_name = rp.get('name', 'autogen') %}
+    {% set duration = rp.get('duration', 'INF') %}
+    {% set replication = rp.get('replication', '1') %}
+    {% set shard_duration = rp.get('shard_duration', False) %}
+    {% set query_rentention_policy = 'RETENTION POLICY {} ON {} DURATION {} REPLICATION {}'.format(
+        rp_name, db.name, duration, replication)
```
Comment (simonpasquier) [inline influxdb/server.sls]: don't you want to manage DEFAULT too?

Comment (SwannCroiset) [reply influxdb/server.sls]: The default PR is named **autogen** with InfluxDB 1.x

File: influxdb/server.sls (hunk 1 above)
Comment (SwannCroiset) [reply influxdb/server.sls]: not especially but if you prefer I'll add one

File: influxdb/server.sls (hunk 2 above)
Comment (SwannCroiset) [reply influxdb/server.sls]: done

File: influxdb/server.sls (hunk 3 above)
Comment (SwannCroiset) [reply influxdb/server.sls]: b/c we cannot add jinja logic inside a jinja **set** directive

File: influxdb/server.sls (hunk 4 above)
Comment (SwannCroiset) [reply influxdb/server.sls]: ok, understood your point

**Review by gthvn1 — APPROVED** [review]

**Comment by SwannCroiset** [comment]
@simonpasquier please review again, I addressed all your points

**Review by simonpasquier — APPROVED** [review]

**Review by elemoine — APPROVED** [review]

**Pull request merged**